server/src/dto/validators/resource_import/versions.ts

```diff
@@ -0,0 +1 @@
+export const versions = [{ value: '2.42.0', schema: import('./2.42.0.json') }];
```

Comment:
Instead of loading all of the schemas in memory, what if we have specific folder for version?

So it will be like `validators/schemas/2.42.0`

and within `2.42.0`
we decouple them into composite schemas ie. main.json, app.json, tooljet_database.json

The advantage here will be that we can directly use specific json file for the specific resource in different dtos and even in frontend

```
validators/
├── schemas/
│   ├── 2.42.0/
│   │   ├── main.json
│   │   ├── app.json
│   │   └── tooljet_database.json
│   ├── 2.43.0/
│   │   ├── main.json
│   │   ├── app.json
│   │   └── tooljet_database.json
│   └── ...
└── validation.ts
```

Reply:
And when the validator runs we load the specific schema for it on demand, something like:

```js
const getSchemaFilePath = (version: string, schemaName: string) => {
  const schemaPath = path.resolve(__dirname, `schemas/${version}/${schemaName}.json`);
  if (fs.existsSync(schemaPath)) {
    return schemaPath;
  }
  return null;
};

const loadSchema = (filePath: string) => {
  return JSON.parse(fs.readFileSync(filePath, 'utf8'));
};

const getSchemaBasedOnAppVersion = async (version: string, schemaName: string) => {
  const schemaFilePath = getSchemaFilePath(version, schemaName);
  if (schemaFilePath) {
    return loadSchema(schemaFilePath);
  }
  return null;
};

```

https://ajv.js.org/guide/combining-schemas.html

Reply:
Please note that when we split this, we can keep the app's section as empty and have the app builder team later add the schema for it.